### ISSUE TITLE
Let use --suffix in a Directory copy/sync without --backup-dir

### ISSUE THREAD

**Issue by naeloob** [rclone/rclone#4339]
#### What is your current rclone version (output from `rclone version`)?
rclone v1.52.0
 - os/arch: windows/amd64
 - go version: go1.14.3

#### What problem are you are trying to solve?
I'm trying to copy directory path tree making a backup of the files updated/deleted on the same dir as the original, not in a backup-dir.
As confirmed in https://github.com/rclone/rclone/issues/4332, actually only works with Backup-dir or copy/sync just a file,

The idea is having : 
```
 -c:/LocalPath
     - File1.txt 1000 Bytes 01/01/2020 15:00
     - File2.txt 2000 Bytes 01/01/2020 15:00
     - File3.txt 3000 Bytes 01/01/2020 15:00
 - CryptRemote:Misc 
     - File1.txt   950 Bytes 01/01/2020 13:00
     - File2.txt 2005 Bytes 01/01/2020 13:00
```
And executing 

`rclone copy c:/LocalPath CryptRemote:Misc --suffix-keep-extension --suffix -BAK -v`

Will left : 
```
 - CryptRemote:Misc 
     - File1-BAK.txt   950 Bytes 01/01/2020 13:00
     - File2-BAK.txt 2005 Bytes 01/01/2020 13:00
     - File1.txt 1000 Bytes 01/01/2020 15:00
     - File2.txt 2000 Bytes 01/01/2020 15:00
     - File3.txt 3000 Bytes 01/01/2020 15:00
```

Bonus Track : 
Adding something like a Filter Rule to choose wich files have to had a Backup and with wich suffix.

```
rclone copy c:/LocalPath CryptRemote:Misc --suffix-keep-extension --suffix "*.{txt,me}<SeparatorToDefine>-BAK" --suffix "*Sample.*<SeparatorToDefine>-OldCopy" -v

or

rclone copy c:/LocalPath CryptRemote:Misc --suffix-keep-extension --suffix "*.{txt,me}""-BAK" --suffix "*Sample.*""-OldCopy" -v

```
Will make a normal copy and a local backup(just if updated or deleted) only of *.txt, *.me files to *-BAK.txt or *-BAK.me and *Sample.* to *Sample-OldCopy.*

Thank you very much in advance.

Regards

**Comment by ncw** [comment]
So what you'd like is for this

> rclone copy c:/LocalPath CryptRemote:Misc --suffix-keep-extension --suffix -BAK -v

To mean this (which doesn't work at the moment - it fails with the error "Failed to copy: destination and parameter to --backup-dir mustn't overlap")

> rclone copy c:/LocalPath CryptRemote:Misc --suffix-keep-extension --suffix -BAK -v --backup-dir CryptRemote:Misc

Is that correct?

I think that would be quite easy.

(Let's start with that first!)

**Comment by naeloob** [comment]
Yes, it could be some type of "Alias" of that.

Thank you!!

**Comment by ncw** [comment]
I just remembered why this is tricky..

When doing a sync rclone needs to ignore all the backup files otherwise it will just delete them.

rsync can put filters in for you automatically but I think this is too much guessing...

So taking your example of `--suffix .BAK --suffix-keep-extension` we need a filter of `--exclude "*.BAK.*"`

Sometimes people use dates in the --suffix, eg `--suffix $(date -I)` that would need a different filter, something like `--exclude "*.[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9].*" so as not to delete this suffix or any old ones.

I could get rclone to check that there is a suitable rule at startup? An exit with an error like this if using `--suffix` with `rclone sync`

    Can't use --suffix without --backup-dir unless there is an --exclude rule to exclude the backups

That check wouldn't be necessary for `rclone copy` or `rclone move`

Is that an acceptable limitation?

**Comment by naeloob** [comment]
For me it's allright. It's a logical limitation.

Maybe just you can let sync using --suffix without --backup-dir warning, with a confirm, that you NEED Exclude Filters to avoid the remove of the Backups, actual and old ones.

**Comment by ncw** [comment]
I've had a go with the code changes here (I haven't done docs or tests)

This gives an error if you attempt an `rclone sync` without a filter to exclude the backup file.

That check isn't needed for `rclone move` or `rclone copy`

Give it a go and see what you think?

https://beta.rclone.org/branch/v1.52.1-069-g5f31c3e3-fix-4339-suffix-beta/ (uploaded in 15-30 mins)

**Comment by naeloob** [comment]
Wuao!!! That was Lightning Fast!!!

Tested on Win64. 
It works like a charm!!

**Comment by naeloob** [comment]
Do you think it will released on the next version?

Regards

**Comment by ncw** [comment]
It will be in 1.53. I need to write the tests and the docs next!

**Comment by naeloob** [comment]
Ok. Thank you!!

**Comment by ncw** [comment]
This work was done as part of de14378734dff98497839529f664e2312511680d which was released in v1.49.0

Namely if you use `--suffix` without `--backup-dir` it will use the destination directory as the --backup dir.

So I'm going to close this.
